src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -264,15 +264,27 @@ public override uint Send(SNIPacket packet)
         /// Receive a packet synchronously
         /// </summary>
         /// <param name="packet">SNI packet</param>
-        /// <param name="timeout">Timeout</param>
+        /// <param name="timeoutInMilliseconds">Timeout in Milliseconds</param>
         /// <returns>SNI error code</returns>
-        public override uint Receive(ref SNIPacket packet, int timeout)
+        public override uint Receive(ref SNIPacket packet, int timeoutInMilliseconds)
         {
             lock (this)
             {
                 try
                 {
-                    _tcpClient.ReceiveTimeout = (timeout != 0) ? timeout : 1;
+                    if (timeoutInMilliseconds < 0)
+                    {   //  infinite timeout (-1) means (0) in TCPClient 
+                        _tcpClient.ReceiveTimeout = 0;
+                    }
+                    else if (timeoutInMilliseconds == 0)
+                    {
+                        // 0 timeout is equal -1 in TCPClient
```

Comment:
This one as well. "For SqlClient (internally) timeout = 0 means that the connection should terminate immediately, this translates to a -1 in TcpClient"